devito/function.py

```diff
@@ -710,12 +735,16 @@ def __init__(self, *args, **kwargs):
             d = Dimension(name='d')
             self.coordinates = Function(name='%s_coords' % self.name,
                                         dimensions=[self.indices[-1], d],
-                                        shape=(self.npoint, self.grid.dim))
+                                        shape=(self.npoint, self.grid.dim),
+                                        space_order=0)
             self._children.append(self.coordinates)
             coordinates = kwargs.get('coordinates', None)
             if coordinates is not None:
                 self.coordinates.data[:] = coordinates[:]
 
+            # A SparseFunction has no halo region
+            self._halo = tuple((0, 0) for i in range(self.ndim))
```

Comment:
Same as above.

Reply:
Oh, so that you can have high order sparse functions?

Reply:
So that you can take derivatives of sparse function. Still trying to find a nice way without overhead to do it.

Reply:
yep, that's what I meant. OK